Source/SwiftyDropbox/Platform/SwiftyDropbox_iOS/OAuthMobile.swift

```diff
@@ -343,10 +345,51 @@ open class MobileSharedApplication: SharedApplication {
     }
 
     open func presentAuthChannel(_ authURL: URL, tryIntercept: @escaping ((URL) -> Bool), cancelHandler: @escaping (() -> Void)) {
-        if let controller = self.controller {
-            let safariViewController = MobileSafariViewController(url: authURL, cancelHandler: cancelHandler)
-            controller.present(safariViewController, animated: true, completion: nil)
+        guard self.authChannel == nil else {
+            return
+        }
+        guard let controller = self.controller else {
+            return
+        }
+        
+        // ASWebAuthenticationSession and SFAuthenticationSession don't work with guided access (rdar://40809553)
+        if !UIAccessibility.isGuidedAccessEnabled {
+            // Sessions don't use openURL for communication like MobileSafariViewController, they just need to be retained
+            if #available(iOS 12.0, *) {
+                let session = MobileWebAuthenticationSession(url: authURL, presentingVC: controller) { [weak self] callbackUrl in
+                    self?.authChannel = nil
+                    if let url = callbackUrl {
+                        DropboxClientsManager.handleRedirectURL(url) { result in }
+                    }
+                }
+                self.authChannel = session
+                session.start()
+                return
+            }
+            if #available(iOS 11.0, *) {
+                let session = MobileAuthenticationSession(url: authURL) { [weak self] callbackUrl in
+                    self?.authChannel = nil
+                    if let url = callbackUrl {
+                        DropboxClientsManager.handleRedirectURL(url) { result in }
+                    }
+                }
+                self.authChannel = session
+                session.start()
+                return
+            }
+        }
+        
+        let safariVC = MobileSafariViewController(url: authURL) { [weak self] svc, didCancel in
+            self?.authChannel = nil
+            
+            if didCancel {
+                cancelHandler()
+            } else {
+                svc.dismiss(animated: true, completion: nil)
+            }
```

Comment:
cancelHandler() goes through the app's openURL flow rather than just dismissing the VC. I don't know why it's doing that but wanted to keep old behavior while also making sure to clear the authChannel variable